### PR TITLE
Add MobileNetwork SDK

### PR DESCRIPTION
**Pull request by andrew-lee-1089**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/mobilenetwork/ci.mgmt.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+pr:
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/servicelinker/ci.mgmt.yml
```
Comment (ArthurMa1978) [inline sdk/mobilenetwork/ci.mgmt.yml:13]: change all `servicelinker` to `mobilenetwork`

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/tests/autorest.tests.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# Generated code configuration
```
Comment (ArthurMa1978) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/tests/autorest.tests.md:1]: remove this file and all generated samples under `tests/Generated/Samples/` folder

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure MobileNetwork management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArthurMa1978) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/README.md:3]: please add introduction of your service here

**Comment by andrew-lee-metaswitch** [comment]
@ArthurMa1978 - this is now ready to review - the only meangiful code i have written is 
azure-sdk-for-net\sdk\mobilenetwork\Azure.ResourceManager.MobileNetwork\tests\MobileNetworkManagementTestBase.cs and azure-sdk-for-net\sdk\mobilenetwork\Azure.ResourceManager.MobileNetwork\tests\Scenario\MobileNetworkTest.cs, everything else is auto-generated.

**Comment by andrew-lee-metaswitch** [comment]
I've done some git rebase magic to make it clear what is auto-gen'd vs what needs review.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/autorest.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Generated code configuration
```
Comment (ArthurMa1978) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/src/autorest.md:1]: Please run `eng\scripts\Export-API.ps1 mobilenetwork` to generate the apiview file which is required for releasing a package.

File: sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
+
+# Microsoft Azure MobileNetwork management client library for .NET
+
+[Azure Private 5G](https://learn.microsoft.com/en-us/azure/private-5g-core/private-5g-core-overview) is an Azure cloud service for deploying and managing 5G core network functions on an Azure Stack Edge device, as part of an on-premises private mobile network for enterprises. The 5G core network functions connect with standard 4G and 5G standalone radio access networks (RANs) to provide high performance, low latency, and secure connectivity for 5G Internet of Things (IoT) devices. Azure Private 5G Core gives enterprises full control and visibility of their private mobile networks.
```
Comment (fengzhou-msft) [inline sdk/mobilenetwork/Azure.ResourceManager.MobileNetwork/README.md:4]: Please remove `en-us` in the `Azure Private 5G` doc link. It's required by [Complianc CI](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2112415&view=logs&j=bc67675d-56bf-581f-e0a2-208848ba68ca&t=480ecc96-5c8f-5bc8-1dbc-f6f32b953559&l=22).

**Comment by fengzhou-msft** [comment]
There are multiple [spellcheck errors](https://dev.azure.com/azure-sdk/public/_build/results?buildId=2112416&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=a880e989-7d1a-5c96-a41f-d540b383cc43&l=143). Please add exemptions in https://github.com/Azure/azure-sdk-for-net/blob/main/.vscode/cspell.json.

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**